ods_ci/tests/Tests/400__ods_dashboard/430__data_science_pipelines/431__data-science-pipelines-api.robot

```diff
@@ -0,0 +1,60 @@
+*** Settings ***
+Documentation       Test suite for OpenShift Pipeline
+Resource            ../../../Resources/RHOSi.resource
+Resource            ../../../Resources/ODS.robot
+Resource            ../../../Resources/Common.robot
+Resource            ../../../Resources/Page/ODH/ODHDashboard/ODHDashboard.robot
+Library             DateTime
+Library             ../../../../libs/DataSciencePipelinesAPI.py
+Suite Setup         Data Science Pipelines Suite Setup
+Suite Teardown      RHOSi Teardown
+
+
+*** Variables ***
+${REDHAT_OPENSHIFT_PIPELINES_YAML}=            tests/Resources/Files/redhat-openshift-pipelines.yaml
+${DATA_SCIENCE_PIPELINES_APPLICATION_YAML}=    tests/Resources/Files/data-science-pipelines-sample.yaml
+${URL_TEST_PIPELINE_RUN_YAML}=                 https://raw.githubusercontent.com/opendatahub-io/data-science-pipelines-operator/main/tests/resources/dsp-operator/test-pipeline-run.yaml
+
+*** Test Cases ***
+Test and end-to-end pipeline workflow via API
+    [Documentation]    Create, run and double check the pipeline result. In the end, clean the pipeline resources.
+    [Tags]    Smoke
+    End to end pipeline workflow via API    ${OCP_ADMIN_USER.USERNAME}    ${OCP_ADMIN_USER.PASSWORD}    myproject1
+    End to end pipeline workflow via API    ${TEST_USER.USERNAME}    ${TEST_USER.PASSWORD}    myproject2
+    # This will fail because there is still a JIRA - RHODS-7615
+    # End to end pipeline workflow via API    ${TEST_USER_3.USERNAME}    ${TEST_USER_3.PASSWORD}    myproject3
+
+*** Keywords ***
+End to end pipeline workflow via API
+    [Documentation]    Create, run and double check the pipeline result. In the end, clean the pipeline resources.
+    [Arguments]     ${username}    ${password}    ${project}
+    Create Pipeline Project    ${project}
+    Install DataSciencePipelinesApplication CR    ${project}
+    ${status}    Login Using User And Password    ${username}    ${password}    ${project}
+    Should Be True    ${status} == 200    DSP routing is working
+    ${pipeline_id}    Create Pipeline    ${URL_TEST_PIPELINE_RUN_YAML}
+    ${run_id}    Create Run    ${pipeline_id}
+    ${run_status}    Check Run Status    ${run_id}
+    Should Be True    '${run_status}' == 'Completed'    Run ends
+    Delete Runs    ${run_id}
+    Delete Pipeline    ${pipeline_id}
```

Comment:
it would be safer to have a Teardown running the deletion, because the teardown is executed even if some of the test keywords fail. While, in this way, if the "Should Be True"  fails, the deletion is not run

Reply:
How can I invoke Teardown with parameters?

Reply:
you can either:
- pass the parameter as a normal keyword when you declare the teardown, like:
```
My Test 1
      [Setup] Setup Keyword
      ${my_pipeline}=   Create Pipeline
      Second Keyword
      ....
      [Teardown]  Clean Up pipeline_to_delete=${my_pipeline}
```
- set some Suite Variables to use in the teardown keyword (I usually do this when using Suite Teardown instead of Test Teardown)

Reply:
This is done